Automattic/wp-calypso
# BackupScheduleSetting: improve BackupScheduled view using useNextBackupSchedule hook

Related to https://github.com/Automattic/jetpack-backup-team/issues/715

This PR improves the precision of the next backup time estimation by leveraging the `useNextBackupSchedule` hook introduced in https://github.com/Automattic/wp-calypso/pull/95602 that returns the next scheduled backup date.

## Proposed Changes

* Removed the previous logic that calculated hours until the next backup using the last backup date and replaced it with precise scheduling data from the `useNextBackupSchedule` hook.
  * If there are any minutes left in the calculation, the hours are rounded up to the next hour.

![CleanShot 2024-10-23 at 20 56 05@2x](https://github.com/user-attachments/assets/46adc748-7c8e-4af8-a2c3-173367070861)

Other minor changes:
* ~Removed `defaultProps` on ActionsButton component, which is now deprecated.~ Update: I was going to remove this, but it seems we have some types issues related to this, so I prefer to address it in a new PR.
* Refactor `useNextBackupSchedule` to named export instead of default export, so we could add more hooks in the same `hooks` file without issues.
* Update schedule backup message in settings page as suggested in p1729507641013709-slack-C07LJAE7AVD

| Scenario | Before | After |
|---|---|---|
| Time set by default | ![CleanShot 2024-10-23 at 21 36 41@2x](https://github.com/user-attachments/assets/13ba7a02-a7a9-4698-b473-94a9bf95a5ad) | ![CleanShot 2024-10-23 at 21 37 07@2x](https://github.com/user-attachments/assets/63c51209-946c-4850-a7da-b483af51536f) |
| Time set by user | ![CleanShot 2024-10-23 at 21 34 07@2x](https://github.com/user-attachments/assets/9962d20d-7b02-4180-869c-480afb6116ad) | ![CleanShot 2024-10-23 at 21 32 16@2x](https://github.com/user-attachments/assets/2296bc67-2719-44af-9429-1dfd8c1b1783) | 

## Why are these changes being made?


The previous logic considered the last completed backup to calculate a relative date, which is not always accurate, especially since we introduced [on-demand backups](https://jetpack.com/blog/on-demand-backups-at-your-fingertips/).

https://github.com/Automattic/jetpack-backup-team/issues/715

## Testing Instructions



* Spin up a Jetpack Cloud live branch.
* Pick a site with Jetpack VaultPress Backup plan with no backups today.
* Navigate to the `Backup` page.
* You should see the `Backup Scheduled` view and the time should be relative to the site scheduled time.
* You can try updating the daily backup time setting and see how it affects the "In the next % hour/s" message.

## Pre-merge Checklist
